mne/channels/layout.py

```diff
@@ -135,6 +154,105 @@ def plot(self, picks=None, show_axes=False, show=True):
+            picks = np.arange(len(self.names))[picks]
+            apply_exclude = False
+        else:
+            try:
+                picks = [_ensure_int(picks)]
+            except TypeError:
+                picks = list(picks) if isinstance(picks, tuple) else deepcopy(picks)
+            apply_exclude = False
+        if apply_exclude:
+            if isinstance(exclude, str):
+                exclude = [exclude]
+            else:
+                try:
+                    exclude = [_ensure_int(exclude)]
+                except TypeError:
+                    exclude = (
+                        list(exclude)
+                        if isinstance(exclude, tuple)
+                        else deepcopy(exclude)
+                    )
+        for var, var_name in ((picks, "picks"), (exclude, "exclude")):
+            if var_name == "exclude" and not apply_exclude:
+                continue
+            if not isinstance(var, (list, tuple, set, np.ndarray)):
+                raise TypeError(
+                    f"'{var_name}' must be a list, tuple, set or ndarray. "
+                    f"Got {type(var)} instead."
+                )
+            for k, elt in enumerate(var):
+                if isinstance(elt, str) and elt in self.names:
+                    var[k] = self.names.index(elt)
+                    continue
+                elif isinstance(elt, str):
+                    raise ValueError(
+                        f"The channel name {elt} provided in {var_name} does not match "
+                        "any channels from the layout."
+                    )
+                try:
+                    var[k] = _ensure_int(elt)
+                except TypeError:
+                    raise TypeError(
+                        f"All elements in '{var_name}' must be integers or strings."
+                    )
+                if not (0 <= var[k] < len(self.names)):
+                    raise ValueError(
+                        f"The value {elt} provided in {var_name} does not match any "
+                        f"channels from the layout. The layout has {len(self.names)} "
+                        "channels."
+                    )
+            if len(var) != len(set(var)):
+                warn(
+                    f"The provided '{var_name}' has duplicates which will be ignored.",
+                    RuntimeWarning,
+                )
+        if apply_exclude:
+            picks = np.array(list(set(picks) - set(exclude)))
+            if len(picks) == 0:
+                raise RuntimeError(
+                    "The channel selection yielded no remaining channels. Please edit "
+                    "the arguments 'picks' and 'exclude' to include at least one "
+                    "channel."
+                )
+        else:
+            picks = np.array(list(set(picks)))
+        self.pos = self.pos[picks]
+        self.ids = self.ids[picks]
+        self.names = [self.names[k] for k in picks]
+        return self
```

Comment:
did you check if this logic is not present elsewhere? I would be surprised if it's not.

Reply:
Yes, the only oneI found that would work is `picks = _picks_to_idx(len(layout.names), picks)` which picks on a number of channels. It's more limited and restrictive on the inputs than the logic above.
I do plan to propose a new channel selection API, to try to 1. clean-up all the redundant code and multiple pick functions in [mne._fiff.pick](https://github.com/mne-tools/mne-python/blob/main/mne/_fiff/pick.py) and 2. open a public API for channel selection (#11913). Hopefully next week 😉